src/openforms/formio/components/vanilla.py

```diff
@@ -338,14 +338,7 @@ class FileSerializer(serializers.Serializer):
     originalName = serializers.CharField(trim_whitespace=False)
     size = serializers.IntegerField(min_value=0)
     storage = serializers.ChoiceField(choices=["url"])
-    type = serializers.CharField(
-        error_messages={
-            "blank": _(
-                "Could not determine the file type. Please make sure the file name "
-                "has an extension."
-            ),
-        }
-    )
+    type = serializers.CharField(required=False, allow_blank=True)
```

Comment:
```suggestion
    type = serializers.CharField(required=True, allow_blank=True)
```

I would definitely still require the key to be present, but we can accept empty strings and process them during validation indeed.